packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -288,6 +291,191 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
+    }
+
+    private NewstartEndRangePanning(shift: number, percentRange: number): number[] {
+        const startRange = this.props.unitController.viewRange.start + (shift * percentRange);
+        const endRange = this.props.unitController.viewRange.end + (shift * percentRange);
+        const arr = [startRange, endRange];
+        return arr;
+    }
+
+    private updatePositionAfterPann(right: number, distance: number){
+        this.positionZoomCenter = this.positionZoomCenter + right * distance;
+        this.positionZoomMouse = this.positionZoomMouse + right * distance;
+    }
+
+    private shift(right: number, distance: number) {
+        const percentRange = this.props.unitController.viewRangeLength * TEN_PERCENT;
+        if (this.NewstartEndRangePanning(right, percentRange)[0] < 0) {
+            this.updatePositionAfterPann(right,distance);
+            this.props.unitController.viewRange = {
+                start: 0,
+                end: this.props.unitController.viewRangeLength
+            };
+        } else if (this.NewstartEndRangePanning(right, percentRange)[1] > this.props.unitController.absoluteRange) {
+            this.updatePositionAfterPann(right,distance);
+            this.props.unitController.viewRange = {
+                start: this.props.unitController.absoluteRange - this.props.unitController.viewRangeLength,
+                end: this.props.unitController.absoluteRange
+            };
+        } else {
+            this.updatePositionAfterPann(right, percentRange);
+            this.props.unitController.viewRange = {
+                start: this.NewstartEndRangePanning(right, percentRange)[0],
+                end: this.NewstartEndRangePanning(right, percentRange)[1]
+            };
+        }
+    }
+
+    private shiftLeft() {
+        const numberLeft = -1;
+        const distance = ( this.props.unitController.viewRange.start);
+        this.shift(numberLeft, distance);
+    }
+
+    private shiftRight() {
+        const numberRight = 1;
+        const distance = (this.props.unitController.absoluteRange - this.props.unitController.viewRange.end);
+        this.shift(numberRight, distance);
+    }
+
+    private wheelEvent(wheel: React.WheelEvent) {
+        if (wheel.shiftKey) {
+            if (wheel.deltaY < 0) {
+                this.shiftLeft();
+            }
+            else if (wheel.deltaY > 0) {
+                this.shiftRight();
+            }
+        }
+    }
+
+    private getPosition(event: React.MouseEvent): number {
+        const offset = this.props.viewRange.getOffset() ?? 0;
+        const scale = this.props.viewRange.getEnd() - this.props.viewRange.getstart();
+        const xPos = this.props.viewRange.getstart() - offset +
+            (event.nativeEvent.offsetX / this.lineChartRef.current.chartInstance.width) * scale;
+        return xPos;
+    }
+
+    private MoveEvent(event: React.MouseEvent) {
+        if (!this.mouseIsDown) {
```

Comment:
Should update the mouse position unconditionally, otherwise for example if you select a range with mouse then W/S, it zooms on the time of the selection start.

I would also only store the mouse x-coordinate, and only calculate the corresponding time when it's used to zoom, that way you don't have to keep updating the mouse 'time' after panning etc..

Reply:
yes I made these changes in another commit "Zoom in and out by click right drag "when I had this problem.